quickwit/quickwit-serve/src/elastic_api_generator.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::{env, path};
+
+use clap::Command;
+use once_cell::sync::Lazy;
+use quickwit_elastic_api_generation::{download_artifacts, generate_api};
+
+pub const ES_ARTIFACTS_DIR_NAME: &str = "elastic-search-artifacts";
+pub const GENERATED_FILE_NAME: &str = "api_specs.rs";
+const GENERATED_FILE_HEADER: &str = r#"
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+/// This file is auto-generated, any change can be overriden.
+"#;
+
+pub static ROOT_DIR: Lazy<path::PathBuf> = Lazy::new(|| {
+    let mf = env::var("CARGO_MANIFEST_DIR").expect("Should be run using 'cargo xtask ...'");
```

Comment:
I don't get the reference to xtask here. Does this PR require xtask to run codegen?

Reply:
yes I dropped xtask